source/cpu/cpu.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cpu
+
+import (
+	"io/ioutil"
+	"path"
+	"regexp"
+
+	"github.com/golang/glog"
+)
+
+// Implement FeatureSource interface
+type Source struct{}
+
+func (s Source) Name() string { return "cpu" }
+
+func (s Source) Discover() ([]string, error) {
+	features := []string{}
+
+	// Check if hyper-threading seems to be enabled
```

Comment:
Seems? Does this mean that this is not a guaranteed method?

Reply:
There is no "guaranteed" method. Even if HT is enabled in HW, it can be effectively soft-disabled (with the same effect as with hard-disable) by offlining all sibling cpus

Reply:
Maybe the wording in the comment could be changed to be more self-evident(?) We just rely on the topology that the kernel sees. If some core has siblings, we deduce that HT seems to be enabled.